plugins/chat/app/services/chat/auto_leave_channels.rb

```diff
@@ -34,13 +36,15 @@ def remove_memberships(params:)
 
       if !group_ids.include?(Group::AUTO_GROUPS[:everyone])
         sql = <<~SQL
+          -- event = #{params.event}
           DELETE FROM user_chat_channel_memberships uccm
           WHERE NOT EXISTS (
             SELECT 1
             FROM group_users gu
             WHERE gu.user_id = uccm.user_id
             AND gu.group_id IN (:group_ids)
           )
+          #{params.user_id ? "AND uccm.user_id = #{params.user_id}" : ""}
```

Comment:
That's the performance fix. When we're "filtering" down by one user, let's only remove their memberships.